src/build/sign.rs

```diff
@@ -1,30 +1,34 @@
 use std::fs;
 use std::fs::DirEntry;
-use std::io::{Error};
+use std::io::Error;
 use std::path::PathBuf;
 
-use crate::error::*;
-
-pub fn copy_sign(folder: &String, entry: &DirEntry, p: &crate::project::Project, version: &String) -> Result<bool, Error> {
+pub fn copy_sign(folder: &String, entry: &DirEntry, p: &crate::project::Project, version: &String, key: &armake2::sign::BIPrivateKey) -> Result<bool, Error> {
     let path = entry.path();
     let cpath = path.clone();
     let cpath = cpath.to_str().unwrap().replace(r#"\"#,"/");
-    let pbo = cpath.replace((folder.clone() + "/").as_str(), "");
-    if !path.ends_with(".pbo") && !pbo.contains(p.prefix.as_str()) {
+    let pbo_path = cpath.replace((folder.clone() + "/").as_str(), "");
+    if !path.ends_with(".pbo") && !pbo_path.contains(p.prefix.as_str()) {
         return Ok(false);
     }
 
     let modname = p.get_modname();
-    fs::copy(&cpath, format!("releases/{}/@{}/{}/{}", version, modname, folder, pbo))?;
+    fs::copy(&cpath, format!("releases/{}/@{}/{}/{}", version, modname, folder, pbo_path))?;
+
+    let signame = p.get_signame(&pbo_path);
+
+    let pbo = armake2::pbo::PBO::read(&mut 
+                                      std::fs::File::open(PathBuf::from(
```

Comment:
What the hell is this indentation! Use variables. :laughing:

Reply:
#38 :(

I'm not used to thinking about indentation. I think Vim probably did it for me. I don't really want to make loads of variables when all it's really doing is updating the PBO path. Rust makes it really verbose to do that 🤷‍♂️

Reply:
It's a lot easier to change variables and move them around than 3x nested function call.